src/moonlink/src/table_handler.rs

```diff
@@ -597,10 +728,17 @@ impl TableHandler {
                                 Err(e) => {
                                     let err = Err(Error::IcebergMessage(format!("Failed to create iceberg snapshot: {e:?}")));
                                     if table_handler_state.has_pending_force_snapshot_request() {
-                                        if let Err(send_err) = table_handler_state.force_snapshot_completion_tx.send(Some(err)) {
+                                        if let Err(send_err) = table_handler_state.force_snapshot_completion_tx.send(Some(err.clone())) {
```

Comment:
TBD: add test case where iceberg snapshot fails for table maintenance result persistence